tests/unittests/cli/test_fast_api.py

```diff
@@ -110,6 +110,7 @@ async def dummy_run_async(
     user_id,
     session_id,
     new_message,
+    state_delta=None,
     run_config: RunConfig = RunConfig(),
 ):
   yield _event_1()
```

Comment:
Maybe just add some custom logic here if state_delta is set, and check the response below?  Then we don't need to use monkeypatch and simplify things.